### PR TITLE
Gps position update

### PR DESCRIPTION
**Pull request by turcato-niccolo**
Hi, I added constructor to GPSPosition that takes a LatLng object

### CODE REVIEW

File: app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import org.mockito.Mockito;
+import org.powermock.api.easymock.PowerMock;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import static org.easymock.EasyMock.expect;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
```
Comment (CremaLuca) [inline app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java]: Codacy Issue found: [Unused import - org.mockito.Mockito.spy.](https://app.codacy.com/manual/CremaLuca/geoCalendar/pullRequest?prid=4964556)

**Comment by CremaLuca** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 20
           

Complexity increasing per file
==============================
- app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java  1
- app/src/main/java/com/eis/geoCalendar/gps/GPSPosition.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/CremaLuca/geoCalendar/pullRequest?prid=4964556&bid=16406630)

File: app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+ * @since 12/28/2019
+ */
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(Location.class)
+public class GPSPositionTest {
+
+    private GPSPosition gpsPosition1;
+    Location location1 = PowerMockito.mock(Location.class);
+    private final double DEF_LATITUDE = 100;
+    private final double DEF_LATITUDE_2 = 101;
```
Comment (CremaLuca) [inline app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java]: Codacy Issue found: [This final field could be made static](https://app.codacy.com/manual/CremaLuca/geoCalendar/pullRequest?prid=4964556)

File: app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+import android.location.Location;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.powermock.api.easymock.PowerMock;
```
Comment (CremaLuca) [inline app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java]: Codacy Issue found: [Unused import - org.powermock.api.easymock.PowerMock.](https://app.codacy.com/manual/CremaLuca/geoCalendar/pullRequest?prid=4964556)

File: app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+    private GPSPosition gpsPosition1;
+    Location location1 = PowerMockito.mock(Location.class);
+    private final double DEF_LATITUDE = 100;
+    private final double DEF_LATITUDE_2 = 101;
+    private final double DEF_LONGITUDE = 100;
+    private final double DEF_LONGITUDE_2 = 101;
+    private final double DELTA = 0;
+    private final double DEF_PRECISION = 0;
+    private final Object GENERIC_OBJ = "ciao";
```
Comment (CremaLuca) [inline app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java]: Codacy Issue found: [This final field could be made static](https://app.codacy.com/manual/CremaLuca/geoCalendar/pullRequest?prid=4964556)

File: app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(Location.class)
+public class GPSPositionTest {
+
+    private GPSPosition gpsPosition1;
+    Location location1 = PowerMockito.mock(Location.class);
+    private final double DEF_LATITUDE = 100;
+    private final double DEF_LATITUDE_2 = 101;
+    private final double DEF_LONGITUDE = 100;
+    private final double DEF_LONGITUDE_2 = 101;
```
Comment (CremaLuca) [inline app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java]: Codacy Issue found: [This final field could be made static](https://app.codacy.com/manual/CremaLuca/geoCalendar/pullRequest?prid=4964556)

File: app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        PowerMockito.doNothing().when(location1).setLongitude(DEF_LONGITUDE);
+        //2nd constructor
+        gpsPosition1.updateLocation(DEF_LATITUDE, DEF_LONGITUDE);
+
+        PowerMockito.when(gpsPosition1.getLatitude()).thenReturn(DEF_LATITUDE);
+        PowerMockito.when(gpsPosition1.getLongitude()).thenReturn(DEF_LONGITUDE);
+    }
+
+    @Test
+    public void getLatitude_tests() {
```
Comment (CremaLuca) [inline app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java]: Codacy Issue found: [The JUnit 4 test method name 'getLatitude_tests' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/manual/CremaLuca/geoCalendar/pullRequest?prid=4964556)

File: app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.powermock.api.easymock.PowerMock;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import static org.easymock.EasyMock.expect;
+import static org.mockito.Mockito.doNothing;
```
Comment (CremaLuca) [inline app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java]: Codacy Issue found: [Unused import - org.mockito.Mockito.doNothing.](https://app.codacy.com/manual/CremaLuca/geoCalendar/pullRequest?prid=4964556)

File: app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.powermock.api.easymock.PowerMock;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import static org.easymock.EasyMock.expect;
```
Comment (CremaLuca) [inline app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java]: Codacy Issue found: [Unused import - org.easymock.EasyMock.expect.](https://app.codacy.com/manual/CremaLuca/geoCalendar/pullRequest?prid=4964556)

File: app/src/main/java/com/eis/geoCalendar/gps/GPSPosition.java [hunk 9]
```diff
@@ -4,49 +4,104 @@
...
+     * @param obj       The reference object with which to compare.
+     * @param precision The precision in meters within which the positions remain equal.
+     * @return true if this object is equal to obj argument in the precision argument range, false otherwise.
+     */
+    public boolean equals(@NonNull Object obj, double precision) {
+        if (obj == null || !(obj instanceof GPSPosition))
+            return false;
+        if (obj == this)
+            return true;
+        if (this.getDistance((GPSPosition) obj) <= precision)
```
Comment (CremaLuca) [inline app/src/main/java/com/eis/geoCalendar/gps/GPSPosition.java]: Codacy Issue found: [Avoid unnecessary if..then..else statements when returning booleans](https://app.codacy.com/manual/CremaLuca/geoCalendar/pullRequest?prid=4964556)

File: app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    @Test
+    public void getLatitude_tests() {
+        Assert.assertEquals(DEF_LATITUDE, gpsPosition1.getLatitude(), DELTA);
+        Assert.assertNotEquals(DEF_LATITUDE_2, gpsPosition1.getLatitude(), DELTA);
+    }
+
+    @Test
+    public void getLongitude_tests() {
```
Comment (CremaLuca) [inline app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java]: Codacy Issue found: [The JUnit 4 test method name 'getLongitude_tests' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/manual/CremaLuca/geoCalendar/pullRequest?prid=4964556)

File: app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+@PrepareForTest(Location.class)
+public class GPSPositionTest {
+
+    private GPSPosition gpsPosition1;
+    Location location1 = PowerMockito.mock(Location.class);
+    private final double DEF_LATITUDE = 100;
+    private final double DEF_LATITUDE_2 = 101;
+    private final double DEF_LONGITUDE = 100;
+    private final double DEF_LONGITUDE_2 = 101;
+    private final double DELTA = 0;
```
Comment (CremaLuca) [inline app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java]: Codacy Issue found: [This final field could be made static](https://app.codacy.com/manual/CremaLuca/geoCalendar/pullRequest?prid=4964556)

File: app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.powermock.api.easymock.PowerMock;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import static org.easymock.EasyMock.expect;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.mock;
```
Comment (CremaLuca) [inline app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java]: Codacy Issue found: [Unused import - org.mockito.Mockito.mock.](https://app.codacy.com/manual/CremaLuca/geoCalendar/pullRequest?prid=4964556)

File: app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+ */
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(Location.class)
+public class GPSPositionTest {
+
+    private GPSPosition gpsPosition1;
+    Location location1 = PowerMockito.mock(Location.class);
+    private final double DEF_LATITUDE = 100;
+    private final double DEF_LATITUDE_2 = 101;
+    private final double DEF_LONGITUDE = 100;
```
Comment (CremaLuca) [inline app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java]: Codacy Issue found: [This final field could be made static](https://app.codacy.com/manual/CremaLuca/geoCalendar/pullRequest?prid=4964556)

File: app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+ * @author Alberto Ursino
+ * @since 12/28/2019
+ */
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(Location.class)
+public class GPSPositionTest {
+
+    private GPSPosition gpsPosition1;
+    Location location1 = PowerMockito.mock(Location.class);
+    private final double DEF_LATITUDE = 100;
```
Comment (CremaLuca) [inline app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java]: Codacy Issue found: [This final field could be made static](https://app.codacy.com/manual/CremaLuca/geoCalendar/pullRequest?prid=4964556)

File: app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    @Test
+    public void getLongitude_tests() {
+        Assert.assertEquals(DEF_LONGITUDE, gpsPosition1.getLongitude(), DELTA);
+        Assert.assertNotEquals(DEF_LONGITUDE_2, gpsPosition1.getLongitude(), DELTA);
+    }
+
+    @Test
+    public void equals_tests() {
```
Comment (CremaLuca) [inline app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java]: Codacy Issue found: [The JUnit 4 test method name 'equals_tests' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/manual/CremaLuca/geoCalendar/pullRequest?prid=4964556)

File: app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java [hunk 16]
```diff
@@ -0,0 +1,77 @@
+package com.eis.geoCalendar.gps;
+
+import android.location.Location;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
```
Comment (CremaLuca) [inline app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java]: Codacy Issue found: [Unused import - org.mockito.Mockito.](https://app.codacy.com/manual/CremaLuca/geoCalendar/pullRequest?prid=4964556)

File: app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java [hunk 17]
```diff
@@ -0,0 +1,77 @@
...
+import org.powermock.api.easymock.PowerMock;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import static org.easymock.EasyMock.expect;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
```
Comment (CremaLuca) [inline app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java]: Codacy Issue found: [Unused import - org.mockito.Mockito.when.](https://app.codacy.com/manual/CremaLuca/geoCalendar/pullRequest?prid=4964556)

File: app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java [hunk 18]
```diff
@@ -0,0 +1,77 @@
+package com.eis.geoCalendar.gps;
+
+import android.location.Location;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
```
Comment (CremaLuca) [inline app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java]: Codacy Issue found: [Unused import - org.mockito.Mock.](https://app.codacy.com/manual/CremaLuca/geoCalendar/pullRequest?prid=4964556)

File: app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java [hunk 19]
```diff
@@ -0,0 +1,77 @@
...
+public class GPSPositionTest {
+
+    private GPSPosition gpsPosition1;
+    Location location1 = PowerMockito.mock(Location.class);
+    private final double DEF_LATITUDE = 100;
+    private final double DEF_LATITUDE_2 = 101;
+    private final double DEF_LONGITUDE = 100;
+    private final double DEF_LONGITUDE_2 = 101;
+    private final double DELTA = 0;
+    private final double DEF_PRECISION = 0;
```
Comment (CremaLuca) [inline app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java]: Codacy Issue found: [This final field could be made static](https://app.codacy.com/manual/CremaLuca/geoCalendar/pullRequest?prid=4964556)

File: app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java [hunk 20]
```diff
@@ -0,0 +1,77 @@
...
+
+    @Test
+    public void getLongitude_tests() {
+        Assert.assertEquals(DEF_LONGITUDE, gpsPosition1.getLongitude(), DELTA);
+        Assert.assertNotEquals(DEF_LONGITUDE_2, gpsPosition1.getLongitude(), DELTA);
+    }
+
+    @Test
+    public void equals_tests() {
+        Assert.assertFalse(gpsPosition1.equals(null));
```
Comment (CremaLuca) [inline app/src/test/java/com/eis/geoCalendar/gps/GPSPositionTest.java]: Codacy Issue found: [Avoid using equals() to compare against null](https://app.codacy.com/manual/CremaLuca/geoCalendar/pullRequest?prid=4964556)

File: app/src/main/java/com/eis/geoCalendar/gps/GPSPosition.java (hunk 9 above)
Comment (turcato-niccolo) [reply app/src/main/java/com/eis/geoCalendar/gps/GPSPosition.java]: Onestamente non vedo il problema

**Comment by CremaLuca** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           

Complexity increasing per file
==============================
- app/src/main/java/com/eis/geoCalendar/gps/GPSPosition.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/CremaLuca/geoCalendar/pullRequest?prid=4964556&bid=16406630)

File: app/src/main/java/com/eis/geoCalendar/gps/GPSPosition.java (hunk 9 above)
Comment (CremaLuca) [inline app/src/main/java/com/eis/geoCalendar/gps/GPSPosition.java]: Codacy Issue found: [Avoid unnecessary if..then..else statements when returning booleans](https://app.codacy.com/manual/CremaLuca/geoCalendar/pullRequest?prid=4964556)

File: .idea/misc.xml [hunk 21]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" project-jdk-name="1.8" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_7" project-jdk-name="1.8" project-jdk-type="JavaSDK">
```
Comment (tonin-alessandra) [inline .idea/misc.xml]: Why this change to JDK 1_7?

**Review by tonin-alessandra — APPROVED** [review]

File: .idea/misc.xml (hunk 21 above)
Comment (turcato-niccolo) [reply .idea/misc.xml]: Android studio doing sh*t

Comment (ricdezen) [reply .idea/misc.xml]: Happens to me too, despite having and using jdk 8

Comment (tonin-alessandra) [reply .idea/misc.xml]: ah ok, don't worry ahaha

Comment (ricdezen) [reply .idea/misc.xml]: https://developer.android.com/studio/write/java8-support
You can give a try to this, worked for me.

Comment (turcato-niccolo) [reply .idea/misc.xml]: This changes the jdk at module level, while the file misc is at project level. Doesn' t work
I just checked, I'm using the android studio embedded jdk, and it's java 8.

Comment (ricdezen) [reply .idea/misc.xml]: My bad it's the project's misc.xml file, didn't notice

Comment (turcato-niccolo) [reply .idea/misc.xml]: Ok it worked now, no clue why😂

Comment (ricdezen) [reply .idea/misc.xml]: Wait what the heck

Comment (turcato-niccolo) [reply .idea/misc.xml]: Dunno, now misc.xml is not showing up because it's not different from its version in develop branch.

**Comment by CremaLuca** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           

Complexity increasing per file
==============================
- app/src/main/java/com/eis/geoCalendar/gps/GPSPosition.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/CremaLuca/geoCalendar/pullRequest?prid=4964556&bid=16406630)

File: app/src/main/java/com/eis/geoCalendar/gps/GPSPosition.java (hunk 9 above)
Comment (CremaLuca) [inline app/src/main/java/com/eis/geoCalendar/gps/GPSPosition.java:93]: Codacy Issue found: [Avoid unnecessary if..then..else statements when returning booleans](https://app.codacy.com/manual/CremaLuca/geoCalendar/pullRequest?prid=4964556)
